### ISSUE TITLE
Refactor Pale Moon to use Places UnifiedComplete

### ISSUE THREAD

**Issue by mattatobin** [MoonchildProductions/UXP#197]
This is our biggest obstacle blocking Pale Moon on UXP. It MUST take top priority.
----
Mozilla has decimated the original implementation for places autocomplete used by the URL bar and other components.. We have **_NO_** choice but to support it in Pale Moon.

Once this is finished we can modify the URL bar back to it's original shape and form.

----

The UnifiedComplete Metabug is here: https://bugzilla.mozilla.org/show_bug.cgi?id=995091

We need the browser parts.. Steal it from Basilisk if necessary.

Blocks #121 #57

**Comment by mattatobin** [comment]
@wolfbeast @janekptacijarabaci @JustOff

**Comment by JustOff** [comment]
This is just beyond my abilities, sorry :confused:

**Comment by mattatobin** [comment]
We just need to find out what bits are needed then do it. SeaMonkey did it.. somehow.. So can we.

**Comment by JustOff** [comment]
@mattatobin, it seems you are now racing against @janekptacijarabaci - https://github.com/janekptacijarabaci/UXP/commits/pm_url_unifiedComplete_1 :horse_racing:

**Comment by mattatobin** [comment]
So it would seem..

**Comment by wolfbeast** [comment]
> PS: I found this just accidentally while fetching UXP updates from all active dev repos.

You know, we do have an issue tracker for a reason where you can indicate an issue you are working on, to prevent this kind of double work done by 2 people...? Not a complaint, just a hint.

**Comment by janekptacijarabaci** [comment]
I'm sorry, but it's just a trial branch (without this tree: https://github.com/MoonchildProductions/UXP/issues/121#issuecomment-381395695 - it doesn't even make sense to try - IMHO it's depends on).

**Comment by mattatobin** [comment]
I have faith in you @janekptacijarabaci You seem to understand this new style of javascript better than most. You can do it. The patches you have been doing have been outstanding and it obvious that your work will make all the difference for us.

**Comment by janekptacijarabaci** [comment]
I'm sorry, but they are big changes.

This tree/branch: https://github.com/MoonchildProductions/UXP/issues/121#issuecomment-381395695 is (AFAIK) ready (at least for testing) but I cannot fully deal with this problem now.
The only thing I can do is test (some) things internally (such as, for example: [tree@pm_url_unifiedComplete_1](https://github.com/janekptacijarabaci/UXP/commits/pm_url_unifiedComplete_1) - but it was just an unsuccessful attempt - nothing else)...

**Comment by mattatobin** [comment]
I'll dig deeper into seeing what can be done on a restoration of the previous functionality in a nonbraking conditional way or go with the final fallback which was the suite option.

We can always look into full on unifiedcomplete later on.

**Comment by mattatobin** [comment]
I have restored nsPlacesAutoComplete and typeAheadResult support in autocomplete.. So that makes this issue WONTFIX. Maybe a future task but we should just use what works even though it needs some style and layout fixups the technology works.

**Comment by wolfbeast** [comment]
@janekptacijarabaci Your attempts at tackling that async mess are greatly appreciated, though! For not you may just want to put it on ice completely, since we'll be using the previous implementation for Pale Moon for the time being.
